### ISSUE TITLE
Beforeall and alfterall tests feature support

### ISSUE THREAD

**Issue by AutomationTester143** [DevExpress/testcafe#5159]
### What is your Test Scenario?
I want to restore the database before all tests of multiple fixtures and drop after all the tests.

### What are you suggesting?
Using this feature, one can simply write code inside it and can perform all the desired configurations and operations before all tests.
I have 5 fixtures that I have written in file .js files. Each fixture has 10 tests and I want to perform database-related configurations before all fixtures or tests

**Comment by ferencbeutel4711** [comment]
this is achievable by using the before() and after() methods of the fixture

**Comment by criskrzysiu** [comment]
no, he don't mean before fixture/after fixture, he man before ALL fixture, after ALL fixture

so it would be adding one more level
for now we have:

1. fixture.before
2. fixture.beforeAll === test.before
3. fixture.afterAll === test.after
4. fixture.aftrer

AutomationTester143  is suggesting adding third level

1. testcafe.before
2. testcafe.beforeAll === fixture.before
3. fixture.beforeAll === test.before
4. fixture.afterAll === test.after
5. testcafe.afterAll ===fixture.after
testcafe.after

Also I would add support for actually setting if for example test.before overwrite fixture.beforeAll or actually extends it (run after)

**Comment by ferencbeutel4711** [comment]
I see. That is possible by creating the test runner programatically. We are also doing that to create some test data for example before starting testcafe. A cleanup is also possible afterwards since the testcafe runner execution is promise-based:

https://devexpress.github.io/testcafe/documentation/reference/testcafe-api/runner/

**Comment by AutomationTester143** [comment]
> no, he don't mean before fixture/after fixture, he mean before ALL fixture, after ALL fixture
> 
> so it would be adding one more level
> for now we have:
> 
> 1. fixture.before
> 2. fixture.beforeAll === test.before
> 3. fixture.afterAll === test.after
> 4. fixture.aftrer
> 
> AutomationTester143 is suggesting adding third level
> 
> 1. testcafe.before
> 2. testcafe.beforeAll === fixture.before
> 3. fixture.beforeAll === test.before
> 4. fixture.afterAll === test.after
> 5. testcafe.afterAll ===fixture.after
>    testcafe.after
> 
> Also I would add support for actually setting if for example test.before overwrite fixture.beforeAll or actually extends it (run after)

Yes I need this feature. Thanks for understanding the need

**Comment by miherlosev** [comment]
Duplicate of https://github.com/DevExpress/testcafe/issues/745
